src/cmds/scripts/pbs_dataservice

```diff
@@ -323,8 +332,23 @@ case "$1" in
 			fi
 			start_dataservice $1
 			ret=$?
-			if [ ${ret} -ne 0 -a "$2" != "PBS" ]; then
-				echo "Failed to start PBS Data Service"
+			if [ ${ret} -ne 0 ]; then
+				if  [ "$2" != "PBS" ]; then
+					echo "Failed to start PBS Data Service"
+				elif [ ${ret} -eq ${version_mismatch} ]; then
```

Comment:
It makes sense to display the logs from pg_ctl before telling the user what it means, so changing the order will not help.

However, I do agree there is some duplication, so I am moving the rm statement outside of all the if conditions as it will handle the scenario when start_dataservice returns 0 as well. Would that work for you?

Reply:
I worked out a way to remove all duplication, updating soon!